testsuite/classlibrary/TestDocument.sc

```diff
@@ -3,6 +3,40 @@ TestDocument : UnitTest {
 	// need a public place to communicate the result
 	classvar <>success;
 
+	test_save_method_writes_file {
+		var doc, file, str, path;
+		if(Platform.ideName == "scqt") {
+			path = Platform.defaultTempDir +/+ "test_document_save.txt";
+			if(File.exists(path)) {
+				this.assert(false, "Document save test file path should not exist: '%'".format(path));
+			} {
+				doc = Document(string: "abcde");
+				// note: AFAIK there is no signal coming from IDE to sclang
+				// to tell it that the document is ready in the editor.
+				// Without a signal, there is no Condition to wait on
+				// so we have no choice but to use a timed wait
+				0.1.wait;
+				doc.save(path);
+				0.1.wait;
+				file = File(path, "r");
+				if(file.isOpen) {
+					protect {
+						str = file.readAllString;
+					} {
+						file.close;
+						File.delete(path);
+						doc.close;
+					};
+					this.assertEquals(str, "abcde", "Document save method should write the correct string to disk");
+				} {
+					this.assert(false, "Document autoRun test could not open '%' for writing".format(path));
+				};
+			}
+		} {
+			// TODO: skip
```

Comment:
what is missing here?

Reply:
Yeah, sorry. Again @jamshark70 wrote this but I think it's a placeholder for other editors. Could be removed now.

Reply:
@jamshark70 Any comments on the queries above. If not I will remove the unused false branch.

Reply:
Actually, as we're inside a running UnitTest, I'm unsure. In the case that we're outside the IDE, the test just shouldn't run. But there's no clear way to assert that the test is just not applicable. What's the correct practice here? Post a warning and do nothing? Assert true with a warning that the test wasn't run? UnitTest doc doesn't make this clear.

Reply:
> What's the correct practice here? Post a warning and do nothing? Assert true with a warning that the test wasn't run? UnitTest doc doesn't make this clear.

We don't have a specific way to address this ATM. For the test runner used in github actions (QPM) we use a script that specifies which tests to exclude and indeed Document tests are [already excluded](https://github.com/supercollider/supercollider/blob/96eaed0b0c7b72e188f000bdacd237e9418ac3b5/testsuite/scripts/gha_test_run_proto.json#L31-L36). I personally think that we should have a built-in test runner if possible where such configuration would be more flexible and allow for conditional exclusion based on which ide is used etc.

Since these tests are currently excluded in the gha test runner, the question remains what should the test do if it's run outside of QPM and not in IDE. IMO it should _not_ assert false... but aside from that I don't know whether it's better to assert true or not and whether to post something or not. I'd be happy to leave this as is. I'd consider adding a little more to the "todo" comment, like `// TODO: decide what to do if the test is run outside of the scide`. I'd be also fine with adding a warning that the test did not run. Unless others feel strongly otherwise, I wouldn't add assert true since we don't have consistent guidelines atm.

Reply:
Great @dyfer! I've added both your TODO text and a warning. Hopefully we can move this along now?